source/adios2/toolkit/interop/hdf5/HDF5Common.cpp

```diff
@@ -54,19 +54,32 @@ HDF5Common::HDF5Common(const bool debugMode) : m_DebugMode(debugMode)
 {
     m_DefH5TypeComplexFloat =
         H5Tcreate(H5T_COMPOUND, sizeof(std::complex<float>));
-    H5Tinsert(m_DefH5TypeComplexFloat, "freal", 0, H5T_NATIVE_FLOAT);
-    H5Tinsert(m_DefH5TypeComplexFloat, "fimg", H5Tget_size(H5T_NATIVE_FLOAT),
+    H5Tinsert(m_DefH5TypeComplexFloat, "r", 0, H5T_NATIVE_FLOAT);
+    H5Tinsert(m_DefH5TypeComplexFloat, "i", H5Tget_size(H5T_NATIVE_FLOAT),
               H5T_NATIVE_FLOAT);
 
     m_DefH5TypeComplexDouble =
         H5Tcreate(H5T_COMPOUND, sizeof(std::complex<double>));
-    H5Tinsert(m_DefH5TypeComplexDouble, "dreal", 0, H5T_NATIVE_DOUBLE);
-    H5Tinsert(m_DefH5TypeComplexDouble, "dimg", H5Tget_size(H5T_NATIVE_DOUBLE),
+    H5Tinsert(m_DefH5TypeComplexDouble, "r", 0, H5T_NATIVE_DOUBLE);
+    H5Tinsert(m_DefH5TypeComplexDouble, "i", H5Tget_size(H5T_NATIVE_DOUBLE),
               H5T_NATIVE_DOUBLE);
 
+    m_DefH5TypeComplexLongDouble =
+        H5Tcreate(H5T_COMPOUND, sizeof(std::complex<long double>));
+    H5Tinsert(m_DefH5TypeComplexLongDouble, "r", 0, H5T_NATIVE_LDOUBLE);
+    H5Tinsert(m_DefH5TypeComplexLongDouble, "i",
+              H5Tget_size(H5T_NATIVE_LDOUBLE), H5T_NATIVE_LDOUBLE);
+
     m_PropertyTxfID = H5Pcreate(H5P_DATASET_XFER);
 }
 
+HDF5Common::~HDF5Common()
+{
+    H5Tclose(m_DefH5TypeComplexLongDouble);
```

Comment:
@guj this seams to write some `H5Tclose(): not a datatype` errors to stderr in some tests.